### PR TITLE
Allow for passing context into the get_image for product variation.

### PR DESCRIPTION
**Pull request by peterfabian**
Problem statement: When the product variation’s image is null, the API response will be set to the same as the parent product, so it will be never null.

p7bje6-2E9-p2

Before writing unit tests, let's discuss if this approach makes sense.

### All Submissions:

* [ ] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/master/.github/CONTRIBUTING.md)?
* [ ] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [ ] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Since get_image behaves differently in edit and view contexts, it is important to pass the context down. In view context, if there is no image set for variation, it returns the parent's image. However, this is wrong for WP admin or mobile app running from the perspective of the admin user.

### How to test the changes in this Pull Request:

1. Create a variation product with 2 variations, assign an image to the parent and to one of the variations.
2. Run a GET on wp-json/wc/v3/productsPARENT_ID/variations/VARIATION_ID_WITHOUT_IMAGE
3. Observe that the image array is populated
4. Apply this patch
4. Run the same request as for 2, response should be the same
4. Run the same request as above, but append `?context=edit` at the end, e.g. `http://one.wordpress.test/wp-json/wc/v3/products/48/variations/62?context=edit`
4. Observe the image is now `null`
4. Try if both contexts (`view` and `edit`) work and return the same response for variation with the image defined.

### Other information:

* [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [ ] Have you successfully run tests with your changes locally?

### Changelog entry

> Fixed image array in `edit` context for product/variations endpoint.

### CODE REVIEW

**Comment by botwoo** [comment]
:bar_chart: Test reports for this pull request have been published and are accessible through the following links:

- [REST API Test Report](https://woocommerce.github.io/woocommerce-test-reports/pr/28498/api/)
- E2E Test Report (TBC)

Latest commit referenced in the reports: `Added changelog.` c5b360e14b76f2f8be5d6fdcdd2a967170352b35
This comment will automatically be updated with the latest referenced commit when you push new changes to this pull request.

---

_Visit the [WooCommerce Test Reports](https://woocommerce.github.io/woocommerce-test-reports) homepage to view all published reports. See the [FAQs](https://github.com/woocommerce/woocommerce-test-reports/wiki/FAQs) page if you're having problems accessing them._

File: plugins/woocommerce/includes/rest-api/Controllers/Version3/class-wc-rest-product-variations-controller.php [hunk 1]
```diff
@@ -33,6 +33,7 @@ class WC_REST_Product_Variations_Controller extends WC_REST_Product_Variations_V
 	 * @return WP_REST_Response
 	 */
 	public function prepare_object_for_response( $object, $request ) {
+		$context  = ! empty( $request['context'] ) ? $request['context'] : 'view';
```
Comment (jorgeatorres) [inline plugins/woocommerce/includes/rest-api/Controllers/Version3/class-wc-rest-product-variations-controller.php]: I'm not sure why our CI didn't pick this up, but this `$context` declaration has one space too many before the `=` sign.

File: plugins/woocommerce/includes/rest-api/Controllers/Version3/class-wc-rest-product-variations-controller.php [hunk 2]
```diff
@@ -33,6 +33,7 @@ class WC_REST_Product_Variations_Controller extends WC_REST_Product_Variations_V
...
 		$data = array(
```
Comment (jorgeatorres) [inline plugins/woocommerce/includes/rest-api/Controllers/Version3/class-wc-rest-product-variations-controller.php]: This `$data` declaration does need some additional spaces so it is aligned with the `$context` one above.

**Review by jorgeatorres — DISMISSED** [review]
Hi @peterfabian! 👋

It makes sense to me that in the `edit` context you'd get the correct image value (i.e. none). I did some testing and this is working perfectly 💯.

While reviewing the code, I noticed a couple of codestyle issues that I've marked in the code. There is, however, something else that I think would be good to fix here but it's not part of your changes: and that is [this conditional check](https://github.com/woocommerce/woocommerce/blob/c5b360e14b76f2f8be5d6fdcdd2a967170352b35/plugins/woocommerce/includes/rest-api/Controllers/Version3/class-wc-rest-product-variations-controller.php#L343) in the code that seems to come from a different time or from a refactor of some kind, as `$image` is never set in the function and thus the conditional check is effectively the same as `if ( true )`. It'd be good to get rid of this unnecessary check in this PR.

Let me know what you think. Thanks!

**Comment by jorgeatorres** [comment]
Hey @peterfabian!

I went ahead and fixed the PHPCS issues mentioned above. Also rebased to sync with latest `trunk`. As mentioned before, this looks good to me, so please let me know if you have any reservations about it. Otherwise, I think we could merge. Thanks!

**Comment by peterfabian** [comment]
Thanks @jorgeatorres . I can't approve my own request, so would you mind giving this an approval so I can merge pls? Thanks!

**Review by jorgeatorres — APPROVED** [review]

**Pull request merged**